tests/local_geometry/test_metric_terms.py

```diff
@@ -201,22 +201,18 @@ def test_k_perp(tmp_path):
     gs2_output = Dataset(gs2_file.with_suffix(".out.nc"))
 
     bunit_over_b0 = pyro.local_geometry.bunit_over_b0
-    theta = gs2_output["theta"][:].data
+    theta_gs2 = gs2_output["theta"][:].data
+    k_perp_gs2 = np.sqrt(gs2_output["kperp2"][0, 0, :].data / 2) / bunit_over_b0
 
-    theta_even = np.linspace(min(theta), max(theta), len(theta))
-    k_perp_gs2_uneven = (
-        gs2_output["kperp2"][0, 0, :].data ** 0.5 / 2**0.5 / bunit_over_b0
-    )
-
-    k_perp_gs2 = np.interp(theta_even, theta, k_perp_gs2_uneven)
-
-    pyro.load_metric_terms(theta=theta_even)
+    pyro.load_metric_terms(ntheta=pyro.numerics.ntheta)
 
     ky = pyro.numerics.ky
-    rho = pyro.local_geometry.rho
+    nperiod = pyro.numerics.nperiod + 1
     theta0 = pyro.numerics.theta0
 
-    k_perp_pyro = pyro.metric_terms.k_perp(ky, rho, theta0)
+    theta_pyro, k_perp_pyro = pyro.metric_terms.k_perp(ky, theta0, nperiod)
+
+    k_perp_gs2 = np.interp(theta_pyro, theta_gs2, k_perp_gs2)
 
     # check within 2%
     assert np.all(np.isclose(k_perp_gs2, k_perp_pyro, rtol=2e-2))
```

Comment:
Could the tolerance be tighter?

Reply:
That's as good as it got. 
![image](https://github.com/pyro-kinetics/pyrokinetics/assets/15210802/865c4979-3585-4dfc-82e0-ca902ff1c082)

Not sure what the differences are but it could some time to work it out. Could drop it to 1.5% and it would pass

Reply:
I could run GS2 with `equal_arc = False`, that might improve it

Reply:
An error of 1.5% seems pretty good for what the test is, but it might be worth seeing if the error drops as the resolution in GS2 is increased. One could increase ntheta to improve the resolution in the calculation _and_ hopefully reduce the interpolation errors, whilst one could increase ntheta_geometry to only improve the resolution in the calculation (i.e. so you could check if it's the interpolation or the calculation at fault).

Reply:
`equal_arc=.false.` made no difference. 

`ntheta_geometry=1024` dropped the difference to <0.7%
![image](https://github.com/pyro-kinetics/pyrokinetics/assets/15210802/e8dd44b5-c199-40c7-88b2-4be83c261636)

`ntheta=64` (from 24) dropped it to <0.2%
![image](https://github.com/pyro-kinetics/pyrokinetics/assets/15210802/1e28659e-d88a-40dc-b465-b7104852a774)


Doing both `ntheta=64` and `ntheta_geometry=1024` 
![image](https://github.com/pyro-kinetics/pyrokinetics/assets/15210802/fa26d6fc-ee03-4c82-ba15-213e5b9bedeb)

Interestingly the increased `ntheta_geometry` drops the difference when |theta| > pi.

I imagine we could keep increasing and get better and better matches. Do you have any preference on the values used? Note this file is also the basic GS2 template so will impact the default settings of other runs.

Reply:
That's perhaps a little bit suspicious -- the error got worse around theta~0 for ntheta=64 going from ntheta_geometry = 64 to ntheta_geometry = 1024, which may suggest we're not converging?

Might it be worth adding in a run with ntheta = 1024?

Of course probably don't want to go down this rabbit hole and <1% error is probably fine. I'd suggest adding ntheta_geometry = something reasonably large in the template as it shouldn't case issues for anything other than ideal_ball (which requires ntheta == ntheta_geometry currently, although I'm hoping to avoid this).